packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -507,56 +486,66 @@ export class ProjectService {
     }
 
     async getJobStatus(jobUuid: string): Promise<Job> {
-        return this.jobModel.getJobstatus(jobUuid);
+        return this.jobModel.get(jobUuid);
     }
 
-    async startJob(projectUuid: string | undefined): Promise<string> {
-        const jobUuid = uuidv4();
-
-        await this.jobModel.upsertJobStatus(
-            jobUuid,
-            projectUuid,
-            JobStatusType.STARTED,
-        );
-
-        return jobUuid;
-    }
-
-    async getAllExplores(
+    async compileProject(
```

Comment:
When manually triggering a recompile:

1. tries to get a project lock (only one process can manually trigger a recompile)
2. if it gets it, create a job, return the job id to the user, continue work in background
3. if it doesn't get the lock, tell the user job is already running